src/spec-lang/tc/semcheck.ts

```diff
@@ -427,3 +432,31 @@ export function scFunctionCall(
 
     throw new Error(`Internal error: NYI semcheck for ${expr.pp()}`);
 }
+
+/**
+ * For Any expression of form old(e1) in `forall(type t in range) e(t)`
+ * throw an error if the expression depends on t.
+ * @param expr
+ * @param ctx
+ * @param typeEnv
+ * @param semMap
+ */
+export function scForAll(expr: SForAll, ctx: SemCtx, typeEnv: TypeEnv, semMap: SemMap): SemInfo {
```

Comment:
So this case worries me a little. It might just work, but I think we should add the following integration tests:

``` forall (uint i in old(a1)) ...```
``` forall (uint i in old(a2[0])) ... ```  (where a2 is a 2 dimensional array)
``` forall (uint i in old(a)) forall (uint j in old(a[i])) ....``` (where a is a 2-dimensional array)

If these compile and run we can leave this without a check. Otherwise we should go back and add checks that the ranges/array are not inside an `old()`.

This is not a blocker to merge this PR.